.github/workflows/rit.yml

```
@@ -23,29 +21,127 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 60
     steps:
+      - name: Checkout Repository # Step needed to access the PR description using github CLI
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
       - name: Set Branch Variables
         id: set-branch-variables
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          github_event_input_powpeg_branch: ${{ github.event.inputs.powpeg-branch }}
+          github_event_input_rit_branch: ${{ github.event.inputs.rit-branch }}
+          github_event_name: ${{ github.event_name }}
+          github_event_pull_request_number: ${{ github.event.pull_request.number }}
+          github_head_ref: ${{ github.head_ref }}
+          github_ref_name: ${{ github.ref_name }}
         run: |
-          # Default values
-          RSKJ_BRANCH="master"
-          RIT_BRANCH="${{ github.event.inputs.rit-branch || 'main' }}"
-          POWPEG_BRANCH="${{ github.event.inputs.powpeg-branch || 'master' }}"
-
-          if [[ "${{ github.event_name }}" == "push" && "${{ github.ref }}" == "refs/heads/*-rc" ]]; then
-            RSKJ_BRANCH="${{ github.ref }}"
-          elif [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
-            RSKJ_BRANCH="${{ github.ref_name }}"
-          elif [[ "${{ github.event_name }}" == "pull_request" ]]; then
-            RSKJ_BRANCH="${{ github.head_ref }}"
+          set -x
+          PR_DESCRIPTION=pr-description.txt
+
+          ALLOWED_BRANCH_CHARACTERS='[-./0-9A-Z_a-z]'
+
+          default_rskj_branch=master
+          default_powpeg_branch=master
+          default_rit_branch=main
+
+          get_branch_from_description()
+          {
+            _prefix=$1
+
+            # On lines matching "`$_prefix:...`", replace the lines with the
+            # thing in ... and print the result.
+            _search_re='\,`'$_prefix:$ALLOWED_BRANCH_CHARACTERS'\{1,\}`'
+            _replace_re='s,.*`'$_prefix:'\('$ALLOWED_BRANCH_CHARACTERS'\{1,\}\)`.*,\1,p'
```

We use `,` inside the patterns, so it's a bad idea to use it to delimiter the regex. 🤦 

```suggestion
            _search_re='\@`'$_prefix:$ALLOWED_BRANCH_CHARACTERS'\{1,\}`@'
            _replace_re='s@.*`'$_prefix:'\('$ALLOWED_BRANCH_CHARACTERS'\{1,\}\)`.*@\1@p'
```